drivers/androidx-sqlite-driver/src/nativeMain/kotlin/app/cash/sqldelight/driver/androidx/sqlite/TransactionsThreadLocal.kt

```diff
@@ -0,0 +1,31 @@
+package app.cash.sqldelight.driver.androidx.sqlite
+
+import app.cash.sqldelight.Transacter
+import kotlin.native.concurrent.Worker
+import kotlinx.atomicfu.atomic
+import kotlinx.atomicfu.update
+
+private class WorkerLocal {
+  val workerData = atomic<MutableMap<Worker, Transacter.Transaction?>>(mutableMapOf())
```

Comment:
TIL about [Kotlin Native WeakReference](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.native.ref/-weak-reference/). Maybe that will help with this?